File: cc/hmap/CheckIfTheSentenceIsPangram.cc
#include <string>

using namespace std;

/**
 * LeetCode 1832. Check if the Sentence Is Pangram
 * https://leetcode.com/problems/check-if-the-sentence-is-pangram/
 */
class CheckIfTheSentenceIsPangram {
public:
    bool checkIfPangram(string sentence) {
        int alphabetMap[26] = {0, };
        int alphabetCount = 0;
        int sLen = sentence.size();
        int idx = 0;
        for (int i = 0; i < sLen; i++) {
            idx = sentence[i] - 'a';
            if (alphabetMap[idx] == 0) {
                alphabetMap[idx]++;
                alphabetCount++;
            }
        }
        return alphabetCount == 26;
    }
};

int main(int argc, char const *argv[]) {
    // ...
    return 0;
}

/* EOF */ 

File: go/hmap/CheckIfTheSentenceIsPangram.go
package main

/**
 * LeetCode 1832. Check if the Sentence Is Pangram
 * https://leetcode.com/problems/check-if-the-sentence-is-pangram/
 */
func checkIfPangram(sentence string) bool {
	alphabetMap := make([]int, 26)
	alphabetCount := 0
	sLen := len(sentence)
	idx := 0
	for i := 0; i < sLen; i++ {
		idx = int(sentence[i] - byte('a'))
		if alphabetMap[idx] == 0 {
			alphabetMap[idx]++
			alphabetCount++
		}
	}
	return alphabetCount == 26
}

func main() {
	// ...
}

// EOF
